geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geotools/SimpleFeaturePropertyAccessor.scala

```diff
@@ -0,0 +1,35 @@
+/***********************************************************************
+* Copyright (c) 2013-2017 Commonwealth Computer Research, Inc.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Apache License, Version 2.0
+* which accompanies this distribution and is available at
+* http://www.opensource.org/licenses/apache2.0.php.
+*************************************************************************/
+
+package org.locationtech.geomesa.utils.geotools
+
+import com.typesafe.scalalogging.LazyLogging
+import org.geotools.filter.expression.{PropertyAccessor, PropertyAccessors}
+import org.opengis.feature.simple.SimpleFeature
+
+import scala.reflect.{ClassTag, _}
+
+object SimpleFeaturePropertyAccessor extends LazyLogging {
+  def getAccessor[T: ClassTag](sf: SimpleFeature, name: String): Option[PropertyAccessor] ={
+    // some mojo to ensure our property accessor is picked up -
+    // our accumulo iterators are not generally available in the system classloader
+    // instead, we can set the context classloader (as that will be checked if set)
+    val contextClassLoader = Thread.currentThread.getContextClassLoader
+    if (contextClassLoader != null) {
+      logger.warn(s"Bypassing context classloader $contextClassLoader for PropertyAccessor loading.")
+    }
+    Thread.currentThread.setContextClassLoader(classTag[T].runtimeClass.getClassLoader)
```

Comment:
Hmm, instead of this we could make this a trait that the classes extend, then change it to `getClass.getClassLoader`. I think I only put the `classOf[Foo].getClassLoader` since it was one less method call.